src/App.tsx

```diff
@@ -8,53 +8,120 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { useEffect, useState } from 'react';
+import { Post } from './types/Post';
+import { getPosts } from './api/api';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [openedPost, setOpenedPost] = useState<Post | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const selectUser = (user: User) => {
+    setOpenedPost(null);
+    setPosts([]);
+    setIsLoading(true);
+    setSelectedUser(user);
+  };
 
-              <Loader />
+  useEffect(() => {
+    if (selectedUser) {
+      getPosts(selectedUser.id)
+        .then(result => {
+          const results = result.map(post => ({
+            id: post.id,
+            userId: post.userId,
+            title: post.title,
+            body: post.body,
+          }));
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+          setPosts(results);
+        })
+        .catch(() => {
+          setIsError(true);
```

Comment:
Consider resetting the `isError` state after a certain period or when a new user is selected. This will improve user experience by clearing the error message once the issue is resolved or a new action is taken.